src/lib/builtins/array.rs

```diff
@@ -470,6 +470,40 @@ pub fn find_index(this: &Value, args: &[Value], interpreter: &mut Interpreter) -
     Ok(Gc::new(ValueData::Number(f64::from(-1))))
 }
 
+/// Array.prototype.fill ( value[, start[, end]] )
+///
+/// The method fills (modifies) all the elements of an array from start index (default 0)
+/// to an end index (default array length) with a static value. It returns the modified array
+/// <https://tc39.es/ecma262/#sec-array.prototype.fill>
+pub fn fill(this: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+    let len: i32 = from_value(this.get_field_slice("length")).expect("Could not get argument");
+    let default_value = Gc::new(ValueData::Undefined);
+    let value = args.get(0).unwrap_or(&default_value);
+    let relative_start = args.get(1).unwrap_or(&default_value).to_num() as i32;
+    let relative_end_val = args.get(2).unwrap_or(&default_value);
+    let relative_end = if relative_end_val.is_undefined() {
+        len
+    } else {
+        relative_end_val.to_num() as i32
+    };
+    let k = if relative_start < 0 {
```

Comment:
Could you give a more descriptive name than `k`?

Reply:
Can you please propose any better name? What do you think about `start`?
The value is named `k` both in tc39 spec and in MDN polyfill example

Reply:
Yeah, `start` is okay.